lib/themes/dosomething/paraneue_dosomething/js/takeover/Takeover.js

```diff
@@ -0,0 +1,245 @@
+
+  spinner: '<div class="spinner"></div>',
+
+  screens: null,
+
+  firstScreen: null,
+
+  lastScreen: null,
+
+  screenIndexes: null,
+
+  /**
+   * Initialize the Signup interface.
+   *
+   * @param  {jQuery} $takeoverContainer the element to add the taked over to.
+   */
+  init: function($takeoverContainer = $('.takeover-container')) {
+    const data = {
+      'header': 'Thanks for signing up',
+    }
+
+    // Add the takeover to the page.
+    const slide = $takeoverContainer.hasClass('-no-slide') ? false : true;
+    this.addTakeover($takeoverContainer, data, slide);
+
+    // Prepare validation fields that got entered into the DOM after page load.
+    Validation.prepareFields($('input'));
+
+    // Set up interactions
+    this.screens = $('.takeover__screen');
+    this.screenIndexes = this.setScreenIndexes();
+    this.firstScreen = this.screens.first();
+    this.lastScreen = this.screens.last();
+    this.firstScreen.addClass('active');
+
+    // Set up submit handlers.
+    this.submitButton = $('.submit-postal-code');
+    this.handleSubmit(this.submitButton, {
+      'button' : this.submitButton,
+      'method' : 'GET',
+      'url'    : 'http://jsonplaceholder.typicode.com/posts/1',
+      'data'   : null,
+    });
+  },
+
+  /**
+   * Adds the takeover to the page with or without a slide animation.
+   *
+   * @param  {jQuery} $container  the container to add the takeover to.
+   * @param  {object} $data       An object containing the data to inject into the templates
+   * @param  {bool}   slide       Whether or not to add the takeover with a slide animation.
+   */
+  addTakeover: function($container, data, slide = false) {
+    const markup = template(takeoverTpl);
+    $container.html(markup(data));
+    this.takeoverContent = $('.takeover');
+
+    if (slide) {
+      this.takeoverContent.hide();
+      this.takeoverContent.delay(500).slideDown('slow', function() {});
+    }
+  },
+
+  /**
+   * Binds click handlers to $button that sends an AJAX request and uses deferred promises to
+   * run callbacks in the order in which they are recieved.
+   *
+   * @param  {jQuery}   $button   Signup buttons
+   * @param  {object}   request   An object that defines the properties we need to make a request.
+   */
+  handleSubmit: function($button, request) {
+    const _this = this;
+
+    // Create custom deferred promise object.
+    let ajaxPromise = $.Deferred().resolve().promise();
+
+    // Bind click handler
+    $button.click(function(e) {
+      e.preventDefault();
+
+      const $this = $(this);
+
+      if (! $('.validation__message.has-error').length) {
+        // Make the ajax call, store the promise.
+        const promise = _this.sendRequest(request);
+
+        // Fire done call backs in the order in which they were recieved.
+        ajaxPromise = ajaxPromise.then(function() {
+          return promise;
+        }).done(function(data) {
+          _this.changeScreens('next');
```

Comment:
I've started using `=> ` arrow functions lately to be able to keep `this` without having to save it in a temporary variable. Totally a stylistic decision, but I feel like it makes things read a bit nicer (plus you can make a lot of things one-liners!)